server/src/main/java/org/elasticsearch/action/search/VersionMismatchException.java

```diff
@@ -0,0 +1,11 @@
+package org.elasticsearch.action.search;
+
+import org.elasticsearch.ElasticsearchException;
+
+public class VersionMismatchException extends ElasticsearchException {
```

Comment:
You also need to declare this exception for serialization in `ElasticsearchException#ElasticsearchExceptionHandle` and add a test in `ExceptionSerializationTests`.